From 9579ee1a5a60b70edf9ffa4a4ff4799d1850d23d Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:11 -0400
Subject: [tools] perf/session: Add missing sample flush for piped events

Message-id: <1408462094-14194-203-git-send-email-jolsa@redhat.com>
Patchwork-id: 88179
O-Subject: [PATCH RHEL7.1 BZ1131394 202/325] perf session: Add missing sample flush for piped events
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 8c16b649606ff9f6d742ad6f71c76fc0ee996c8e
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Oct 18 15:29:02 2013 +0300

description
===========
Piped events can be sorted so a final flush is needed.

Add that and remove a redundant 'err = 0'.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/session.c b/tools/perf/util/session.c
index d974208..429b1f6 100644
--- a/tools/perf/util/session.c
+++ b/tools/perf/util/session.c
@@ -1268,7 +1268,9 @@ more:
  if (!session_done())
   goto more;
 done:
- err = 0;
+ /* do the final flush for ordered samples */
+ self->ordered_samples.next_flush = ULLONG_MAX;
+ err = flush_sample_queue(self, tool);
 out_err:
  free(buf);
  perf_session__warn_about_errors(self, tool);
@@ -1397,13 +1399,13 @@ more:
         "Processing events...");
  }
 
- err = 0;
  if (session_done())
-  goto out_err;
+  goto out;
 
  if (file_pos < file_size)
   goto more;
 
+out:
  /* do the final flush for ordered samples */
  session->ordered_samples.next_flush = ULLONG_MAX;
  err = flush_sample_queue(session, tool);
-- 
1.7.1